src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -62,6 +73,25 @@ trait RegularSync {
       }
   }
 
+  private def insertMinedBlock(block: Block, parentTd: BigInt) = {
+    val result: Either[BlockExecutionError, Seq[Receipt]] = ledger.executeBlock(block, blockchainStorages, validators)
```

Comment:
when executing block in prepare we do not perform full validation because there is no nonce/mixHash so I decided to execute it again to check consistency and to not be force to cache and pass around mpt hash
so: we can, but it would make code more complicated
I am not convinced about performance gain